src/test/java/stashpullrequestbuilder/stashpullrequestbuilder/StashBuildListenerTest.java

```diff
@@ -0,0 +1,84 @@
+package stashpullrequestbuilder.stashpullrequestbuilder;
+
+import static org.mockito.Mockito.anyString;
+import static org.mockito.Mockito.eq;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import hudson.model.FreeStyleBuild;
+import hudson.model.TaskListener;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.mockito.junit.MockitoRule;
+import org.mockito.quality.Strictness;
+
+@RunWith(MockitoJUnitRunner.class)
+public class StashBuildListenerTest {
+
+  @Rule public JenkinsRule jenkinsRule = new JenkinsRule();
+  @Rule public MockitoRule rule = MockitoJUnit.rule().strictness(Strictness.STRICT_STUBS);
+
+  private StashBuildListener stashBuildListener;
+
+  private StashCause stashCause;
+  private TaskListener taskListener;
+
+  @Mock private FreeStyleBuild build;
+
+  @Before
+  public void before() throws Exception {
+    stashBuildListener = new StashBuildListener();
+    taskListener = jenkinsRule.createTaskListener();
+    stashCause =
+        new StashCause(
+            "StashHost",
+            "SourceBranch",
+            "TargetBranch",
+            "SourceRepositoryOwner",
+            "SourceRepositoryName",
+            "PullRequestId",
+            "DestinationRepositoryOwner",
+            "DestinationRepositoryName",
+            "PullRequestTitle",
+            "SourceCommitHash",
+            "DestinationCommitHash",
+            "BuildStartCommentId",
+            "PullRequestVersion",
+            null);
+  }
+
+  @Test
+  public void onStartedChangesBuildDescription() throws Exception {
+    when(build.getCause(eq(StashCause.class))).thenReturn(stashCause);
+
+    stashBuildListener.onStarted(build, taskListener);
+
+    verify(build, times(1)).setDescription(anyString());
+  }
+
+  @Test
+  public void onStartedIgnoredWrongCause() throws Exception {
```

Comment:
can you use a more verbose name? Something like "onStarted will not set description if there is no stash cause"

Reply:
as an exception to the general rule I think it's actually better to use snake_case for test case names

Reply:
Good point, it will allow us to make CamelCase names stand out.

I wish I could just write test description in comments and have some low-key or even autogenerated test names that would not duplicate the comments. But I cannot think of a good implementation, and I don't see good ideas on StackOverflow: https://stackoverflow.com/questions/706354/

Reply:
I'm find with just using snake case method names. What is the problem with it?

Reply:
I think snake case is the best that we can do with Java. I remember seeing a test framework where test cases had free style names with spaces, but I think it was for Ruby. I cannot find it at the moment.

Reply:
Maye you are thinking about Spock (Groovy)? I wouldn't want Spock just for that though